### PR TITLE
Various centos-ruby image improvements

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

**Comment by mfojtik** [comment]
@smarterclayton can you please review this one?

I like the idea of @bparees usage script, also added more dependencies for @jwforres

**Comment by mfojtik** [comment]
fyi, with these changes in place, I was able to successfully STI build and run empty Rails application.

```
sti build https://github.com/mfojtik/rails-app-example openshift/centos-ruby-extended rails-app --debug
```

```
 ~/code/rh/go-sti → docker run --rm -i -p 9292:9292 -t rails-app
Puma 2.8.2 starting...
* Min threads: 0, max threads: 16
* Environment: development
* Listening on tcp://0.0.0.0:9292
172.17.42.1 - - [28/Apr/2014 11:36:34] "GET / HTTP/1.1" 200 - 0.7453
```

File: centos-ruby/extended/Dockerfile [hunk 1]
```diff
@@ -7,7 +7,38 @@
 FROM openshift/centos-ruby
 MAINTAINER Michal Fojtik <mfojtik@redhat.com>
 
-RUN yum install --assumeyes gcc automake autoconf curl-devel openssl-devel \
+# OpenShift RHEL6 repository to provide additional rubygems and
+# libraries for Ruby developers
+#
+ADD https://mirror.openshift.com/pub/openshift-origin/nightly/rhel-6/dependencies/openshift-rhel6-dependencies.repo /etc/yum.repos.d/openshift-rhel6-dependencies.repo
```
Comment (smarterclayton) [inline centos-ruby/extended/Dockerfile]: I'm not sure this is appropriate in this context.  If we add this here we're implicitly saying that we will have this repo available for a very long time.  Is that the case?

**Comment by smarterclayton** [comment]
Just the one comment

File: centos-ruby/extended/Dockerfile (hunk 1 above)
Comment (mfojtik) [inline centos-ruby/extended/Dockerfile]: After discussion with Troy on Friday, it seems like we are going to keep this repo for a while, as it provides the source of missing RPM's for various packages in CentOS. Especially for ruby193-*.

File: centos-ruby/base/etc/sdk [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Puma web server is available when using the 'extended' layer,
```
Comment (smarterclayton) [inline centos-ruby/base/etc/sdk]: Nat sure this comment is valid

Comment (mfojtik) [inline centos-ruby/base/etc/sdk]: Right, it is available if you put it into Gemfile.

Comment (smarterclayton) [inline centos-ruby/base/etc/sdk]: But it's not included in the extended layer, only as part of the gem.

Comment (mfojtik) [inline centos-ruby/base/etc/sdk]: @smarterclayton fixed.

**Pull request merged**